src/einsteinpy/symbolic/ricci.py

```diff
@@ -1,5 +1,91 @@
+import numpy as np
+import sympy
+
 from .tensor import Tensor
+from .metric import MetricTensor
 
 
 class RicciTensor(Tensor):
-    pass
+    """
+    Class for defining Ricci Tensor
+    """
+
+    def __init__(self, arr, syms):
+        """
+        Constructor and Initializer
+
+        Parameters
+        ----------
+        arr : ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray or list
+            Sympy Array or multi-dimensional list containing Sympy Expressions
+        syms : tuple or list
+            Tuple of crucial symbols dentoting time-axis, 1st, 2nd, and 3rd axis (t,x1,x2,x3)
+
+        Raises
+        ------
+        TypeError
+            Raised when arr is not a list or sympy Array
+        TypeError
+            syms is not a list or tuple
+
+        """
+        super(RicciTensor, self).__init__(arr)
+        if isinstance(syms, (list, tuple)):
+            self.syms = syms
+            self.dims = len(self.syms)
+        else:
+            raise TypeError("syms should be a list or tuple")
+
+    @classmethod
+    def from_christoffels(cls, chris):
+        """
+        Get Ricci Tensor calculated from a Christoffel Symbols
+
+        Parameters
+        ----------
+        chris : ~einsteinpy.symbolic.ricci.RicciTensor
+            Christoffel Symbols from which Ricci Tensor to be calculated
+
+        """
+        arr, syms = chris.tensor(), chris.symbols()
+        dims = len(syms)
+        ricci_list = (np.zeros(shape=(dims, dims, dims, dims), dtype=int)).tolist()
+        for i in range(dims ** 4):
```

Comment:
Ricci Tensor is contraction of riemann Tensor over 1st and 3rd indices
 R<sub>ij</sub> = R<sup>k</sup><sub>ikj</sub> where is follows einstein notation and is summed over index `k`.
What I suggest that you could calculate Riemann Tensor(from the previously created class) and from that calculate Ricci Tensor

Reply:
sympy natively supports tensor contraction! Yayy!
https://docs.sympy.org/latest/modules/tensor/array.html#products-and-contractions.

Now one of the classmethods can look like:
```
@classmethod
def from_riemann(cls, riemann):
    return cls(sympy.tensorcontraction(riemann.tensor(), (0, 2)), riemann.syms)

```

for other classmethods like `from_christoffels` and `from_metric`, we can obtain Riemann Tensor first and then can pass it to `from_riemann` method mentioned above.